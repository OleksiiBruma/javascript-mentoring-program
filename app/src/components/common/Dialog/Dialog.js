import React, { useState } from "react";
import {
  Button,
  DialogActions,
  DialogTitle,
  Dialog as MUIDialog
} from "@material-ui/core";

const Dialog = ({ handleConfirm, text }) => {
  const [open, setOpen] = useState(false);
  const handleClickOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  return (
    <div>
      <Button size="small" color="primary" onClick={handleClickOpen}>
        {text}
      </Button>
      <MUIDialog open={open} onClose={handleClose}>
        <DialogTitle>{"Are you sure?"}</DialogTitle>
        <DialogActions>
          <Button autoFocus onClick={handleConfirm} color="primary">
            Sure
          </Button>
          <Button autoFocus onClick={handleClose} color="primary">
            Cancel
          </Button>
        </DialogActions>
      </MUIDialog>
    </div>
  );
};

export default Dialog;
