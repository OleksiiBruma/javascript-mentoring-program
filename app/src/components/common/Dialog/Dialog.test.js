import renderer from "react-test-renderer";
import React from "react";
import Dialog from "../Dialog/Dialog";

it("Dialog should handle open confirm and close", () => {
  const handlerClick = () => true;

  const component = renderer.create(
    <Dialog handleConfirm={() => handlerClick(id)} text={"Delete"} />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
