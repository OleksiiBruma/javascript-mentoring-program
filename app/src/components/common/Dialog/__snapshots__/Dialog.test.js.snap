// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should handle open confirm and close 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Delete
    </span>
  </button>
</div>
`;
